script/job_generator/generate_job_yaml.py

```diff
@@ -113,7 +113,7 @@ def job_tags(job: Dict[str, Tuple[str, str]]) -> List[str]:
         and job[ALPAKA_ACC_GPU_HIP_ENABLE][VERSION] != OFF
     ):
         return ["x86_64", "rocm"]
-    return ["x86_64", "cpuonly"]
+    return ["cpuonly"]
```

Comment:
I think this is wrong. If you add only the `cpuonly` tag, the GitLab CI will decide, if a job is running on x86 or ARM. This means also, if you restart a failed job it is possible, that the job failed on x86 and is restarted on a ARM runner. At the moment, the function is not used, so it will not affect the CI. Also, the job generator needs an extension, to decide if a job is executed on a x86 CPU or a ARM CPU (it is required, that you can implement filter rules like clang x is not supporting ARM).

Because this is bigger change, I suggest to revert the change and add a todo: `# TODO(j-stephan): extend cpuonly tags for the ARM runner`

Reply:
Does it matter whether we run on x86 or ARM? We also do not differentiate between Intel and AMD CPUs, or what kind of instruction sets the CPU support.

Reply:
No, that's not true. In practice, there is platform depending code, which is not so well tested on ARM because of the much smaller user base. 

I have some personal experience with my Raspberry PI. And it is still a problem. I had the case this weekend, that a flashing tool does not worked on the PI but on my x86 laptop. And the flashing tool should not use processor specific functions, such like vector commands.

Reply:
We should explicitly test ARM and x86. One problem is that we need different base containers, at least if we use the pre built containers @SimeonEhrig is shipping.
The pre-built containers are required to reduce the runtime of a single job.

Reply:
In general, I am against supporting ARM as long as we have no access to an ARM systems to reproduce issues.

Reply:
We have access to an A64fx system (well, I do).

Reply:
But we can also pin our jobs to a specific architecture if that is the broader consensus. Or just support x86 for the time being. We are somewhat democratic so this is a majority decision.